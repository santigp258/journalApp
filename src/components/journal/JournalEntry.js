import React from "react";

export const JournalEntry = () => {
  return (
    <div className="journal__entry mt-1">
      <div
        className="journal__entry-picture"
        style={{
          backgroundSize: "cover",
          backgroundImage:
            "url(https://upload.wikimedia.org/wikipedia/commons/thumb/b/b6/Image_created_with_a_mobile_phone.png/1200px-Image_created_with_a_mobile_phone.png)",
        }}
      ></div>

      <div className="journal__entry-body">
          <p className="journal__entry-title">
              A new day
          </p>
          <div className="journal__entry-content">
              Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ipsum veniam repellendus velit eaque tempora et illo doloremque nostrum esse eius.
          </div>
      </div>
      <div className="journal__entry-date-box">
          <span>Monday</span>
            <h4>28</h4>          
      </div>
    </div>
  );
};
