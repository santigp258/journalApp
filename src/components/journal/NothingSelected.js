import React from 'react'

export const NothingSelected = () => {
    return (
        <div className="nothing__main-content">
            <p className="animate__animated animate__flipInX">
                Select Something. Create an entry!
            </p>
            <i className="far fa-star fa-4x mt-5 animate__animated animate__backInRight"></i>
        </div>
    )
}
