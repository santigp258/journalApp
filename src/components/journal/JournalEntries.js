import React from "react";
import { useSelector } from "react-redux";
import { JournalEntry } from "./JournalEntry";

export const JournalEntries = () => {
  const { notes: entries } = useSelector((state) => state.notes);
  return (
    <div className="journal__entries">
      {entries &&
        entries.map((entry) => <JournalEntry key={entry.id} {...entry} />)}
    </div>
  );
};
