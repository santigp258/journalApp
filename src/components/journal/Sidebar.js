import React from "react";
import { JournalEntries } from "./JournalEntries";

export const Sidebar = () => {
  return (
    <aside className="journal__sidebar">
      <div className="journal__sidebar-navbar">
        <h3 className="mt-5">
          <i className="far fa-moon"></i>
          <span>Santiago</span>
        </h3>

        <button className="btn">Logout</button>
      </div>

      <div className="journal__new-entry">
        <i className="fa fa-calendar-plus fa-5x"></i>
        <p>New Entry</p>
      </div>

      <JournalEntries/>
    </aside>
  );
};
