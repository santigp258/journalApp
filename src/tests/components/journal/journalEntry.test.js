import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router";
import { JournalEntry } from "../../../components/journal/JournalEntry";
const middlewares = [thunk];

//create store
const mockStore = configureStore(middlewares);

const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn();
const note = {
  id: 123,
  date: 0,
  title: "hola",
  body: "mundo",
  url: "https://hixd.com/foto.png",
};
const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <JournalEntry {...note} />
    </MemoryRouter>
  </Provider>
);

describe("pruebas en <JournalEntry/>", () => {
  test("debe mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe  de activar la nota", () => {
    wrapper.find(".journal__entry").prop("onClick")();

    expect(store.dispatch).toHaveBeenCalled();
  });
});
