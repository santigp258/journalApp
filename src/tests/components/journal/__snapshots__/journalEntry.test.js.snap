// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <JournalEntry/> debe mostrarse correctamente 1`] = `
<div
  className="animate__animated animate__pulse journal__entry mt-1"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(https://hixd.com/foto.png)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      hola
    </p>
    <div
      className="journal__entry-content"
    >
      mundo
    </div>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
